### ISSUE TITLE
Style/ReturnNilInPredicateMethodDefinition is firing incorrectly

### ISSUE THREAD

**Issue by tomhughes** [rubocop/rubocop#12068]
## Expected behavior

`Style/ReturnNilInPredicateMethodDefinition` should only fire when a predicate method returns `nil`.

## Actual behavior

Offences are reported for code that uses `nil` but does not return it:

```
Offenses:

app/models/oauth_token.rb:51:23: C: [Correctable] Style/ReturnNilInPredicateMethodDefinition: Return false instead of nil in predicate methods.
    invalidated_at != nil
                      ^^^
lib/bounding_box.rb:92:19: C: [Correctable] Style/ReturnNilInPredicateMethodDefinition: Return false instead of nil in predicate methods.
    to_a.exclude?(nil)
                  ^^^

2 files inspected, 2 offenses detected, 2 offenses autocorrectable
Finished in 0.807309715077281 seconds
```

## Steps to reproduce the problem

Run rubocop on these methods:

```ruby
def invalidated?
  invalidated_at != nil
end

def complete?
  to_a.exclude?(nil)
end
```

## RuboCop version

```
1.55.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-capybara 2.18.0
  - rubocop-factory_bot 2.23.1
  - rubocop-minitest 0.31.0
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
```
